patsy/util.py

```diff
@@ -720,3 +720,23 @@ def test_safe_string_eq():
         assert safe_string_eq(unicode("foo"), "foo")
 
     assert not safe_string_eq(np.empty((2, 2)), "foo")
+
+def check_pickle_version(version, required_version, name=""):
+    if version > required_version:
+        error_msg = "This version of patsy is too old to load this pickle"
+    elif version < required_version:
+        error_msg = "This pickle is too old and not supported by this version of patsy anymore"
+    else:
+        return
```

Comment:
Presumably we'll eventually want to support multiple pickle versions at the same time -- but I guess we can always refactor this when the time comes :-)